keps/sig-instrumentation/3077-contextual-logging/README.md

```diff
@@ -0,0 +1,965 @@
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+No.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Previous state is irrelevant, so a previous rollback has no effect.
+
+###### Are there any tests for feature enablement/disablement?
+
+Not applicable.
+
+### Rollout, Upgrade and Rollback Planning
+
+Nothing special needed. The same logging flags as before will be supported.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The worst case would be that a null logger instance somehow gets passed into a
+function and then causes log calls to crash. The design of the APIs makes that
+very unlikely and code reviews should be able to catch the code that causes
+this.
+
+###### What specific metrics should inform a rollback?
+
+Components start to crash with a nil pointer panic in the `logr` package.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Not applicable.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+By design usage of the feature is not observable at runtime.
+
+###### How can someone using this feature know that it is working for their instance?
+
+Only by inspecting the source code and checking whether `klog` or `logr.Logger`
+is used for log messages.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Performance should be similar to logging through klog, with some overhead for
+passing around a logger.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Other
+  - Details: CPU resource utilization before/after an upgrade
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Counting log calls and their kind (through klog vs. logger, `Info`
+vs. `Error`) would be possible, but then cause overhead by itself with
+questionable usefulness.
+
+### Dependencies
+
+None.
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+Same as before.
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+Pod scheduling (= "startup latency of schedulable stateless pods" SLI) might
```

Comment:
why? because of the answer to the next question?

Reply:
Yes.

I looked at existing SLIs and that might be one that is sensitive to CPU consumption.